sklearn/conftest.py

```diff
@@ -50,6 +50,13 @@ def global_dtype(request):
     yield request.param
 
 
+@pytest.fixture(
+    params=[np.float16, pytest.param(np.float32, marks=_SKIP32_MARK), np.float64]
+)
+def float_dtype(request):
+    yield request.param
+
+
```

Comment:
I'm not sure a new fixture for this specific case is the best solution. I was imagining just a single test explicitly testing `np.float16` _in addition_ to the one parameterized by the `global_dtype` fixture. What do you think @glemaitre?